### ISSUE TITLE
google_checks: False positive on left curly brace in switch statement with lambda-like construct

### ISSUE THREAD

**Issue by bradleylarrick** [checkstyle/checkstyle#15851]
```
$ export JAVA_HOME=~/.jdks/corretto-21.0.3/
$ PATH=$JAVA_HOME/bin:$PATH
$ javac Test.java 

$ cat Test.java 
class Test {
  public void foo() {
    int i = 0;
    switch (i) {
      case 1 -> { // one octet remaining; add two bits and load 2 quintets
        i++;
      }
      default -> {} // Will never get here
    }
  }
}

$ java -jar checkstyle-10.20.0-all.jar -c /google_checks.xml Test.java 
Starting audit...
[WARN] /var/tmp/Test.java:5:17: '{' at column 17 should be on a new line. [LeftCurlyNl]
Audit done.

$ java -jar google-java-format-1.23.0-all-deps.jar Test.java > TestUpdated.java
$ diff -u Test.java TestUpdated.java 
# no diff
```

Expected no violations.

------

I found a couple more false positives on the LeftCurlyNL in a switch statement with lambda-like construct. I formatted the code using google-java-format-1.24.0-all-deps.jar and ran checkstyle using checkstyle-10.20.0-all.jar. The warnings seem to be generated when the left curly brace is followed by a comment:
````
   123      if (result.length > rpos) {
   124        switch (modulus) {
   125          case 1 -> { // one octet remaining; add two bits and load 2 quintets
   126            workarea = workarea << 2;
   127            rpos = loadCodes(workarea, result, rpos, 2);
   128          }
   129          case 2 -> { // two octet remaining; add four bits and load 4 quintets
   130            workarea = workarea << 4;
   131            rpos = loadCodes(workarea, result, rpos, 4);
   132          }
   133          case 3 -> { // three octets remaining; add one bit and load 5 quintets
   134            workarea = workarea << 1;
   135            rpos = loadCodes(workarea, result, rpos, 5);
   136          }
   137          case 4 -> { // four octets remaining; add 3 bits and load 7 quintets
   138            workarea = workarea << 3;
   139            rpos = loadCodes(workarea, result, rpos, 7);
   140          }
   141          default -> {} // Will never get here
   142        }
   143      }
````
generated the following:
````
[WARN] C:\projects\commons\src\main\java\org\natuna\commons\Base32.java:125:19: '{' at column 19 should be on a new line. [LeftCurlyNl]
[WARN] C:\projects\commons\src\main\java\org\natuna\commons\Base32.java:129:19: '{' at column 19 should be on a new line. [LeftCurlyNl]
[WARN] C:\projects\commons\src\main\java\org\natuna\commons\Base32.java:133:19: '{' at column 19 should be on a new line. [LeftCurlyNl]
[WARN] C:\projects\commons\src\main\java\org\natuna\commons\Base32.java:137:19: '{' at column 19 should be on a new line. [LeftCurlyNl]
````
And:
````
   246      switch (modulus) {
   247        case 0 -> { // no leftovers
   248        }
   249        case 1, // 5 bits remaining; shouldn't occur from valid encoding, so ignore
   250                2 -> // 10 bits remaining; drop two and output one octet
   251            result[pos++] = (byte) ((workarea >> 2) & MASK_8BITS);
   252        case 3 -> // 15 bits remaining; shouldn't occur from valid encoding
   253            result[pos++] = (byte) ((workarea >> 7) & MASK_8BITS);
   254        case 4 -> // 20 bits remaining; drop the last four and load 2 bytes
   255            pos = loadBytes(workarea >> 4, result, pos, 2);
   256        case 5 -> // 25 bits remaining; drop the last bit and load 3 bytes
   257            pos = loadBytes(workarea >> 1, result, pos, 3);
   258        case 6 -> // 30 bits remaining; shouldn't occur from valid encoding
   259            pos = loadBytes(workarea >> 6, result, pos, 3);
   260        case 7 -> // 35 bits remaining; drop last three and load 4 bytes
   261            pos = loadBytes(workarea >> 3, result, pos, 4);
   262        default -> { // Will never get here
   263        }
   264      }
````
Generated:
````
[WARN] C:\projects\commons\src\main\java\org\natuna\commons\Base32.java:247:17: '{' at column 17 should be on a new line. [LeftCurlyNl]
[WARN] C:\projects\commons\src\main\java\org\natuna\commons\Base32.java:262:18: '{' at column 18 should be on a new line. [LeftCurlyNl]
````

**Comment by romani** [comment]
@Zopsss , please glance.

**Comment by romani** [comment]
```
$ cat Test.java 
class Test {
  public void foo() {
    int i = 0;
    switch (i) {
      case 1 -> {
        i++;
      }
      default -> {}
    }
  }
}
$ java -jar google-java-format-1.23.0-all-deps.jar Test.java > TestUpdated.java
$ diff -u Test.java TestUpdated.java
$
```



This shows that case/default are considered as regular blocks of class/if-else/ ...
```
$ cat Test.java 
class Test {
  public void foo() {
    int i = 0;
    switch (i) {
      case 1 ->
        {
          i++;
        }
      default -> {

      }
    }
  }
}

$ java -jar google-java-format-1.23.0-all-deps.jar Test.java > TestUpdated.java
$ diff -u Test.java TestUpdated.java
--- Test.java	2024-11-06 04:40:51.754540844 -0800
+++ TestUpdated.java	2024-11-06 04:41:05.210022844 -0800
@@ -2,13 +2,10 @@
   public void foo() {
     int i = 0;
     switch (i) {
-      case 1 ->
-        {
-          i++;
-        }
-      default -> {
-
+      case 1 -> {
+        i++;
       }
+      default -> {}
     }
   }
 }
```

**Comment by Zopsss** [comment]
@romani your PR is merged, and user's problem is resolved. The issue is ready to be closed.

**Comment by romani** [comment]
@bradleylarrick , fix is released, please update your project.
We appreciate your feedback on how it works.

**Comment by aleuner** [comment]
Found the same problem with 10.20.0 last week (didn't check earlier versions), had to workaround due to the enhanced switch statements.
Problem is indeed fixed for me with 10.20.1. I no longer need any workarounds for the supplied `google_checks.xml`.

Thanks @romani for fixing this, thanks @bradleylarrick for reporting!
